oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsPowerSource.java

```diff
@@ -151,7 +151,7 @@ private static WindowsPowerSource getPowerSource(String name) {
 
         HANDLE hdev = SetupApi.INSTANCE.SetupDiGetClassDevs(GUID_DEVCLASS_BATTERY, null, null,
                 SetupApi.DIGCF_PRESENT | SetupApi.DIGCF_DEVICEINTERFACE);
-        if (WinBase.INVALID_HANDLE_VALUE != hdev) {
+        if (!WinBase.INVALID_HANDLE_VALUE.equals(hdev)) {
```

Comment:
Maybe also here, swap the order around so it follows the rest?
```suggestion
        if (!hdev.equals(WinBase.INVALID_HANDLE_VALUE)) {
```

Reply:
yes, also if for some reason `hdev` for some unexpected reason turns out to be `null` it prevents a NullPointerException

Reply:
As @mprins says, it's generally preferred to call `.equals()` from a known non-null constant and that would be preferred here.   One can go read the documentation for [SetupDiGetClassDevs()](https://docs.microsoft.com/en-us/windows/win32/api/setupapi/nf-setupapi-setupdigetclassdevsw) to verify that it doesn't return null so in this case it's fine; but generally better to be defensive!

Reply:
(So in this case it's better as is and the others should be switched to match.)

Reply:
Thanks for the details. Kept this as in and switched others